### PR TITLE
FE-137: Subaccount IP pools dropdown shows "default" when ip_pool is unset, does not allow you to clear the field

### PR DESCRIPTION
**Pull request by bkemper**
What to test?
- Confirm a subaccount can be created without an IP Pool
- Confirm a subaccount can be updated to add an IP Pool
- Confirm a subaccount can be updated to switch an IP Pool
- Confirm a subaccount can be updated to remove the IP Pool
- Confirm a subaccount can be created or updated with 'default' IP Pool

### CODE REVIEW

File: src/actions/helpers/subaccounts.js [hunk 1]
```diff
@@ -34,10 +35,9 @@ export function formatSubaccount(values, getState) {
     subaccount.setup_api_key = false;
   }
 
-  if (ipPool !== 'default') {
+  if (restrictedToIpPool) {
```
Comment (bkemper) [inline src/actions/helpers/subaccounts.js]: I'm not happy with our form data spilling out into actions.  I'm going to file a ticket to refactor.

File: src/pages/subaccounts/components/IpPoolSelect.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  }));
+
+  return (
+    <Field
+      name="ipPool"
+      component={SelectWrapper}
+      label="IP Pool"
+      options={formattedOptions}
+      disabled={disabled}
+      placeholder=" "
```
Comment (bkemper) [inline src/pages/subaccounts/components/IpPoolSelect.js]: - [ ] Remove placeholder.

File: src/pages/subaccounts/components/IpPoolSelect.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  const formattedOptions = options.map(({ id, name }) => ({
+    label: `${name} (id: ${id})`,
+    value: id
+  }));
+
+  return (
+    <Field
+      name="ipPool"
+      component={SelectWrapper}
+      label="IP Pool"
```
Comment (jonambas) [inline src/pages/subaccounts/components/IpPoolSelect.js]: `Select an IP pool`

Also, maybe also add helptext that links to the create ip pool page - 
> 'Need a new IP pool? Create a new one"

File: src/pages/subaccounts/components/RestrictToIpPoolCheckbox.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+import { Field } from 'redux-form';
+import CheckboxWrapper from 'src/components/reduxFormWrappers/CheckboxWrapper';
+
+export default function RestrictToIpPoolCheckbox({ disabled }) {
+  return (
+    <Field
+      name="restrictedToIpPool"
+      component={CheckboxWrapper}
+      type="checkbox"
+      label="Restrict traffic to a single IP Pool"
```
Comment (jonambas) [inline src/pages/subaccounts/components/RestrictToIpPoolCheckbox.js]: > Restrict this subaccount's mail deliveries to a single IP pool

File: src/pages/subaccounts/components/SubaccountCreateForm.js [hunk 5]
```diff
@@ -50,12 +55,12 @@ const mapStateToProps = (state, props) => ({
...
   showGrants: valueSelector(state, 'grantsRadio') === 'select',
-  ipPools: state.ipPools.list,
+  ipPools: getNonDefaultIpPools(state),
+  restrictedToIpPool: valueSelector(state, 'restrictedToIpPool'),
   initialValues: {
     grants: getInitialSubaccountGrants(state),
     grantsRadio: 'all',
-    createApiKey: true,
-    ipPool: 'default'
+    createApiKey: true
```
Comment (jonambas) [inline src/pages/subaccounts/components/SubaccountCreateForm.js]: should add `restrictedToIpPool: false`

File: src/pages/subaccounts/components/RestrictToIpPoolCheckbox.js (hunk 4 above)
Comment (jonambas) [reply src/pages/subaccounts/components/RestrictToIpPoolCheckbox.js]: Maybe we can use helptext here to describe the unchecked behavior

> If left unchecked, this subaccount may send using any IP pool

File: src/pages/subaccounts/components/EditTab.js [hunk 6]
```diff
@@ -12,10 +12,15 @@ import SubaccountEditForm from './SubaccountEditForm';
 
 export class EditTab extends Component {
 
-  onSubmit = ({ name, status, ipPool }) => {
+  onSubmit = ({ ipPool, name, restrictedToIpPool, status }) => {
     const { editSubaccount, subaccount, getSubaccount, showAlert } = this.props;
+    const nextSubaccount = {
+      name,
+      status,
+      ip_pool: restrictedToIpPool ? ipPool : '' // ugh, not present doesn't work to unset
```
Comment (jonambas) [inline src/pages/subaccounts/components/EditTab.js]: Maybe change comment to: 'pass an empty string to unset ip pool'

File: src/pages/subaccounts/components/SubaccountCreateForm.js [hunk 7]
```diff
@@ -50,12 +55,12 @@ const mapStateToProps = (state, props) => ({
...
   }
 });
```
Comment (jonambas) [inline src/pages/subaccounts/components/SubaccountCreateForm.js]: I have no idea how this and the edit form are working without `enableReinitialize: true` /shrug

File: src/pages/subaccounts/components/SubaccountCreateForm.js [hunk 8]
```diff
@@ -28,14 +32,15 @@ export class SubaccountCreateForm extends Component {
...
-            <IpPoolSelect ipPools={ipPools} disabled={submitting} />
+            <RestrictToIpPoolCheckbox disabled={submitting} />
+            {restrictedToIpPool && <IpPoolSelect disabled={submitting} options={ipPools} />}
           </Panel.Section>
-        }
+        )}
         <Panel.Section>
           <Button submit primary disabled={submitting || pristine}>
-            { submitting ? 'Loading...' : 'Create' }
+            {submitting ? 'Loading...' : 'Create'}
```
Comment (jonambas) [inline src/pages/subaccounts/components/SubaccountCreateForm.js]: `Create Subaccount`

File: src/pages/subaccounts/components/SubaccountEditForm.js [hunk 9]
```diff
@@ -15,37 +18,43 @@ export class SubaccountEditForm extends Component {
...
+        {Boolean(ipPools.length) && (
+          <Panel.Section>
+            <RestrictToIpPoolCheckbox disabled={submitting || compliance} />
+            {restrictedToIpPool && <IpPoolSelect disabled={submitting || compliance} options={ipPools} />}
+          </Panel.Section>
+        )}
         <Panel.Section>
           <Button submit primary disabled={pristine || submitting || compliance}>
-            { submitting ? 'Updating...' : 'Update' }
+            {submitting ? 'Updating...' : 'Update'}
```
Comment (jonambas) [inline src/pages/subaccounts/components/SubaccountEditForm.js]: `Update Subaccount`

File: src/pages/subaccounts/components/EditTab.js [hunk 10]
```diff
@@ -12,10 +12,15 @@ import SubaccountEditForm from './SubaccountEditForm';
...
+    };
 
-    return editSubaccount(subaccount.id, { name, status, ip_pool: ipPool }).then(() => {
+    return editSubaccount(subaccount.id, nextSubaccount).then(() => {
       showAlert({ type: 'success', message: 'Updated subaccount' });
```
[the remark this answers is not in the document]
Comment (bkemper) [reply src/pages/subaccounts/components/EditTab.js]: Yep, I'll add it to the refactor ticket.

File: src/pages/subaccounts/components/IpPoolSelect.js (hunk 3 above)
Comment (bkemper) [reply src/pages/subaccounts/components/IpPoolSelect.js]: That's a good idea.  I'll cut a ticket for that.

Comment (bkemper) [reply src/pages/subaccounts/components/IpPoolSelect.js]: https://jira.int.messagesystems.com/browse/FE-457

Comment (jonambas) [reply src/pages/subaccounts/components/IpPoolSelect.js]: Why a new ticket?

Comment (bkemper) [reply src/pages/subaccounts/components/IpPoolSelect.js]: It felt weird adding a feature in a defect and I hope it leads to a larger discussion about how to nest these forms.

**Pull request merged**